GuessingGame/Service.cs

```diff
@@ -3,71 +3,75 @@
+                MoneyToBet();
+            }
+            return amount;
+        }
+
+        public int GuessNumber()
+        {
+            Console.Write("Enter the number to guess: ");
+            if (!int.TryParse(Console.ReadLine(), out int number))
+            {
+                Console.WriteLine("Invalid input. Please enter a valid whole number.");
+            }
 
-    public string GetLevel(){
-        Console.Write(" Choose level (Easy/Medium/Hard) : ");
-        string difficulty = Console.ReadLine();
-        
-                    switch (difficulty.ToLower())
-                    {
-                    case "easy":
-                    return GetResult(1,5,5);
-                    case "medium":
-                    return GetResult(1,10,10);  
-                    case "hard":
-                    return GetResult(1,15,15);
-                    default:
+            return number;
+        }
+
+        public string ChooseDifficultyLevelWhichReturnMinMaxMutipler(int amount,int guessNumber)
+        {
+            Console.Write(" Choose level (Easy/Medium/Hard) : ");
+            string difficulty = Console.ReadLine();
+
+            switch (difficulty.ToLower())
+            {
+                case "easy":
+                    return GuessingGameResult(1, 5, 5,amount,guessNumber);
+                case "medium":
+                    return GuessingGameResult(1, 10, 10,amount,guessNumber);
+                case "hard":
+                    return GuessingGameResult(1, 15, 15,amount,guessNumber);
+                default:
                     Console.WriteLine("enter a correct string");
-                        return GetLevel();
-                       
-                    }
+                    return ChooseDifficultyLevelWhichReturnMinMaxMutipler(amount,guessNumber);
+            }
 
-    }
-    public string GetResult(int minRange,int maxRange,int multiplier){
-                    Random random = new Random();
-                    int randomNumber = random.Next(minRange, maxRange);
+        }
+        public string GuessingGameResult(int minRange, int maxRange, int multiplier,int amount,int guessNumber)
+        {
 
-                    int game = GetAmount() * multiplier;
+            int randomNumber = _numberGenerator.Generate(minRange, maxRange);
 
-                    if (GetNumber() == randomNumber)
-                    {
-                        Console.WriteLine($"Congratulations! You won {game} dollars.");
-                    }
-                    else
-                    {
-                        Console.WriteLine($"Sorry, you lost. The correct number is {randomNumber}.");
-                    }
-        Console.WriteLine($"Let's Start Once More");
-        return GetLevel();
+            int betAmount =  amount * multiplier;
+
+            if (guessNumber == randomNumber)
+            {
+                Console.WriteLine($"You won {betAmount} dollars.");
```

Comment:
For the next exersise, it would be good to see if the text here is tested in the unit test. Right now the result you are testing against is different to the unit test. ie the user will see a different response to what your method returns. For the lambda exersise it should be easier because you won't have a Console.WriteLine but this would have been easy to do outside this method anyway.